tests/baselines/reference/destructuringTuple.types

```diff
@@ -23,20 +23,20 @@ declare var receiver: typeof tuple;
 // Repros from #32140
 
 const [oops1] = [1, 2, 3].reduce((accu, el) => accu.concat(el), []);
->oops1 : any
->[1, 2, 3].reduce((accu, el) => accu.concat(el), []) : number
+>oops1 : never
+>[1, 2, 3].reduce((accu, el) => accu.concat(el), []) : never[]

```

Comment:
We used to think `[1, 2, 3].reduce((accu, el) => accu.concat(el), [])` is a _number_. You probably need the (unfortunate) overload resolution error in the function body to get that far off track in the first place, but there’s really no reason why destructuring `[oops1]` from the result should have any effect on what we think this type is.